src/system/dev/dev_driver.c

```diff
@@ -128,10 +128,59 @@ const struct fs_driver _dev_driver = {.close_r = dev_close_r,
                                       .lseek_r = dev_lseek_r,
                                       .read_r = dev_read_r,
                                       .write_r = dev_write_r,
+                                      .mkdir_r = dev_mkdir_r,
+                                      .link_r = dev_link_r,
+                                      .unlink_r = dev_unlink_r,
+                                      .chdir_r = dev_chdir_r,
+                                      .chmod_r = dev_chmod_r,
+                                      .pathconf_r = dev_pathconf_r,
+                                      .getcwd_r = dev_getcwd_r,
+                                      .stat_r = dev_stat_r,
                                       .ctl = dev_ctl};
 
 const struct fs_driver* const dev_driver = &_dev_driver;
 
+int dev_mkdir_r(struct _reent* r, const char* path, mode_t mode) {
```

Comment:
I know it's annoying without a compiler to check this for you, but this function doesn't match the type you have in the struct. I haven't checked the others, but can you as well?

Reply:
These should go with the other newlib driver functions earlier in this page. At the very least, all these functions need to go before their respective struct.

Reply:
Done